hooks/use-sound.ts

```diff
@@ -0,0 +1,103 @@
+'use client';
+import type {State} from '@app/config/state';
+import {useCallback, useContext, useEffect, useState} from 'react';
+import {Context} from '../contexts/audio';
+
+const useSound = (state: State): void => {
+	const [loadedSounds, setLoadedSounds] = useState<{[url: string]: AudioBuffer}>({});
```

Comment:
You can use an aliased record type here, which is easier to read than an inline declaration.

```suggestion
type AudioCache = Record<string, AudioBuffer>;

const useSound = (state: State): void => {
	const [loadedSounds, setLoadedSounds] = useState<AudioCache>({});
```